test/integration/master/audit_dynamic_test.go

```diff
@@ -0,0 +1,293 @@
+
+	t.Run("one sink", func(t *testing.T) {
+		_, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Create(&sink1)
+		require.NoError(t, err, "failed to create audit sink1")
+		t.Log("created audit sink1")
+
+		var finalErr error
+		err = wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+			// reset event list
+			eventList1.Store(auditinternal.EventList{})
+
+			// perform configmap ops
+			configMapOperations(t, kubeclient)
+
+			// check for corresponding events
+			missing, err := checkForEvents(t, eventList1)
+			if err != nil {
+				finalErr = fmt.Errorf("%v: failed to match all expected events for sink1, events %#v not found", err, missing)
+				return false, nil
+			}
+
+			return true, nil
+		})
+		require.NoError(t, err, finalErr)
+	})
+
+	t.Run("two sink", func(t *testing.T) {
+		_, err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Create(&sink2)
+		require.NoError(t, err, "failed to create audit sink2")
+		t.Log("created audit sink2")
+
+		var finalErr error
+		err = wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+			// reset event list
+			eventList1.Store(auditinternal.EventList{})
+			eventList2.Store(auditinternal.EventList{})
+
+			// perform configmap ops
+			configMapOperations(t, kubeclient)
+
+			// check for corresponding events in both sinks
+			missing, err := checkForEvents(t, eventList1)
+			if err != nil {
+				finalErr = fmt.Errorf("%v: failed to match all expected events for sink1, events %#v not found", err, missing)
+				return false, nil
+			}
+
+			missing, err = checkForEvents(t, eventList2)
+			if err != nil {
+				finalErr = fmt.Errorf("%v: failed to match all expected events for sink1, events %#v not found", err, missing)
+				return false, nil
+			}
+
+			return true, nil
+		})
+		require.NoError(t, err, finalErr)
+	})
+
+	t.Run("delete sink", func(t *testing.T) {
+		err := kubeclient.AuditregistrationV1alpha1().AuditSinks().Delete("test2", &metav1.DeleteOptions{})
+		require.NoError(t, err, "failed to delete audit sink2")
+		t.Log("deleted audit sink2")
+
+		var finalErr error
+		err = wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+			// reset event lists
+			eventList1.Store(auditinternal.EventList{})
+			eventList2.Store(auditinternal.EventList{})
+
+			// perform configmap ops
+			configMapOperations(t, kubeclient)
+
+			// check for corresponding events in sink1
+			missing, err := checkForEvents(t, eventList1)
+			if err != nil {
+				finalErr = fmt.Errorf("%v: failed to match all expected events for sink1, events %#v not found", err, missing)
+				return false, nil
+			}
+
+			// check that sink2 is empty
+			if len(eventList2.Load().(auditinternal.EventList).Items) != 0 {
+				finalErr = fmt.Errorf("eventList2 should be empty")
+				return false, nil
+			}
+			return true, nil
+		})
+		require.NoError(t, err, finalErr)
+	})
+
+	t.Run("update sink", func(t *testing.T) {
```

Comment:
I'm not used to seeing t.Run used serially like this... the tests seem fairly interdependent. if one fails won't the following ones fail too?

Reply:
yeah they would, its more just a way to segment the individual pieces, we could remove it if it feels like its complicating things

Reply:
I think I'll write a wrapper here that checks if the test passed and if not fail the parent